src/main/java/com/example/MachineService/controllers/MachineController.java

```diff
@@ -0,0 +1,167 @@
+package com.example.MachineService.controllers;
+
+import com.example.MachineService.entities.Machine;
+import com.example.MachineService.entities.User;
+import com.example.MachineService.services.MachineService;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+@RequestMapping("/tenant")
+@Controller
+public class MachineController {
+
+    private MachineService machineService;
+
+    public MachineController(MachineService machineService) {
+        this.machineService = machineService;
+    }
+
+    //@GetMapping("/{userId}/machine/device/{deviceId}")
+    @GetMapping("/{userId}/machine/device/add")
+    public String addMachineRequest(Model model, @PathVariable Long userId)
+    {
+        if(!machineService.isUserIn(userId))
+        {
+            model.addAttribute("msg","user Id is invalid");
+            return "DisplayMsg";
+        }
+        model.addAttribute("Machine", new Machine());
+        model.addAttribute("userId", userId);
+        return "AddDevice"; //add machine form
+    }
+
+    @PostMapping("/{userId}/machine/device/add")
```

Comment:
remove "add" word from the path as when the method is post then it should be add API